src/features/leverage-tokens/hooks/useMinSharesGuard.ts

```diff
@@ -0,0 +1,106 @@
+import { useEffect, useState } from 'react'
+
+interface UseMinSharesGuardParams<TPlan> {
+  currentStep: string
+  plan: TPlan | undefined
+  getMinValue: (plan: TPlan) => bigint | undefined
```

Comment:
Stricter typing okay into the below. Let's get the return types from the plans as explicit types.

```ts
plan: MintPlan | RedeemPlan | undefined
```